tests/terratest/functions/set/setRKE1.go

```diff
@@ -0,0 +1,163 @@
+		ec2ConfigBlock := nodeTemplateBlockBody.AppendNewBlock("amazonec2_config", nil)
+		ec2ConfigBlockBody := ec2ConfigBlock.Body()
+		ec2ConfigBlockBody.SetAttributeValue("access_key", cty.StringVal(terraformConfig.AWSAccessKey))
+		ec2ConfigBlockBody.SetAttributeValue("secret_key", cty.StringVal(terraformConfig.AWSSecretKey))
+		ec2ConfigBlockBody.SetAttributeValue("ami", cty.StringVal(terraformConfig.Ami))
+		ec2ConfigBlockBody.SetAttributeValue("region", cty.StringVal(terraformConfig.Region))
+		awsSecGroupNames := format.ListOfStrings(terraformConfig.AWSSecurityGroupNames)
+		ec2ConfigBlockBody.SetAttributeRaw("security_group", awsSecGroupNames)
+		ec2ConfigBlockBody.SetAttributeValue("subnet_id", cty.StringVal(terraformConfig.AWSSubnetID))
+		ec2ConfigBlockBody.SetAttributeValue("vpc_id", cty.StringVal(terraformConfig.AWSVpcID))
+		ec2ConfigBlockBody.SetAttributeValue("zone", cty.StringVal(terraformConfig.AWSZoneLetter))
+		ec2ConfigBlockBody.SetAttributeValue("root_size", cty.NumberIntVal(terraformConfig.AWSRootSize))
+		ec2ConfigBlockBody.SetAttributeValue("instance_type", cty.StringVal(terraformConfig.AWSInstanceType))
+	}
+
+	if terraformConfig.Module == "linode_rke1" {
+		linodeConfigBlock := nodeTemplateBlockBody.AppendNewBlock("linode_config", nil)
+		linodeConfigBlockBody := linodeConfigBlock.Body()
+		linodeConfigBlockBody.SetAttributeValue("image", cty.StringVal(terraformConfig.LinodeImage))
+		linodeConfigBlockBody.SetAttributeValue("region", cty.StringVal(terraformConfig.Region))
+		linodeConfigBlockBody.SetAttributeValue("root_pass", cty.StringVal(terraformConfig.LinodeRootPass))
+		linodeConfigBlockBody.SetAttributeValue("token", cty.StringVal(terraformConfig.LinodeToken))
+	}
+
+	rootBody.AppendNewline()
+
+	// Resource cluster
+	clusterBlock := rootBody.AppendNewBlock("resource", []string{"rancher2_cluster", "rancher2_cluster"})
+	clusterBlockBody := clusterBlock.Body()
+	dependsOnTemp := hclwrite.Tokens{
+		{Type: hclsyntax.TokenIdent, Bytes: []byte(`[rancher2_node_template.rancher2_node_template]`)},
+	}
+	clusterBlockBody.SetAttributeRaw("depends_on", dependsOnTemp)
+	clusterBlockBody.SetAttributeValue("name", cty.StringVal(terraformConfig.ClusterName))
+	rkeConfigBlock := clusterBlockBody.AppendNewBlock("rke_config", nil)
+	rkeConfigBlockBody := rkeConfigBlock.Body()
+	rkeConfigBlockBody.SetAttributeValue("kubernetes_version", cty.StringVal(k8sVersion))
+	networkBlock := rkeConfigBlockBody.AppendNewBlock("network", nil)
+	networkBlockBody := networkBlock.Body()
+	networkBlockBody.SetAttributeValue("plugin", cty.StringVal(terraformConfig.NetworkPlugin))
+	rootBody.AppendNewline()
+
+	// Resource node pools
+	clusterSyncNodePoolIDs := ``
+	num := 1
+	for _, pool := range nodePools {
+		poolNum := strconv.Itoa(num)
+		if !pool.Etcd && !pool.Cp && !pool.Wkr {
+			return false, fmt.Errorf(`no roles selected for pool` + poolNum + `; at least one role is required`)
+		}
+		if pool.Quantity <= 0 {
```

Comment:
[scalability enhancement]:
since we reuse this across multiple providers, it might be worth breaking these out into its own `common`-like file that is called by each `set<provider>`.go file. Maybe something like `setResourceChecks.go` or something.

Reply:
Agreed ^

Reply:
Sure thing, yeah that's a great suggestion. I'll break the provider sections out into their own files

Reply:
@slickwarren Actually, thinking about this more, these provider blocks are specific to rke1, so although there are different providers, they will not be reused w/ other cluster types (rke2/k3s etc).  I don't see value in breaking these sections out + away from the setRKE1.go file.  With these blocks being specific to RKE1, are you still requesting I break these out into separate files??  I think that might cause clutter, and things are easily found ATM, but if this will be easier for the team to manage moving forward, this is a simple fix that could be implemented.  Looking for further insights, if you still would agree this should be broken out?

Reply:
@slickwarren Actually, thinking about this more, these provider blocks are specific to rke1, so although there are different providers, they will not be reused w/ other cluster types (rke2/k3s etc).  I don't see value in breaking these sections out + away from the setRKE1.go file.  With these blocks being specific to RKE1, are you still requesting I break these out into separate files??  I think that might cause clutter, and things are easily found ATM, but if this will be easier for the team to manage moving forward, this is a simple fix that could be implemented.  Looking for further insights, if you still would agree this should be broken out?

Reply:
its already reused several times here, right? So moving forward if we have a common file for this, we can just specify that this is for rke1, then add more helpful functions once rke2/k3s/etc are added.